contentpacks/khanacademy.py

```diff
@@ -194,74 +196,258 @@ def retrieve_translations(crowdin_project_name, crowdin_secret_key, lang_code="e
+
+    # Convert all keys of nodes to snake case from camel case.
+    for key in node_data:
+        node_data[key] = convert_all_nodes_to_camel_case(node_data[key])
+
+    # Remove any topic nodes that are hidden, deleted, or set to 'do_not_publish'
+    # Also remove those flags from the nodes themselves.
+
+    topic_nodes = []
+
+    for node in node_data["topics"]:
+        hidden = node.pop("hide")
+        dnp = node.pop("do_not_publish")
+        deleted = node.pop("deleted")
+        # We want to remove all of these, except the root node,
+        # the only node we do hide, but we use for defining the overall KA channel
+        if not (hidden or dnp or deleted) or node.get("id") == "x00000000":
+            topic_nodes.append(node)
+
+    node_data["topics"] = topic_nodes
+
+    # Hack to hardcode the mp4 format flag on Videos.
+    for node in node_data["videos"]:
+        node["format"] = "mp4"
+
+    # Flatten node_data
+
+    node_data = [node for node_list in node_data.values() for node in node_list]
+
+    # Modify slugs by kind to give more readable URLs
+
+    node_data = modify_slugs(node_data)
+
+    # Remove blacklisted items
+
+    node_data = apply_black_list(node_data)
+
+    # Create paths, deduplicate slugs, remove orphaned content and childless topics
+
+    node_data = create_paths_remove_orphans_and_empty_topics(node_data)
+
+    # Modify id keys to match KA Lite id formats
+
+    node_data = modify_ids(node_data)
+
+    # Save node_data to disk
+
+    with open(path, "w") as f:
+        return ujson.dump(node_data, f)
+
+
+topic_attributes = [
+    'childData',
+    'deleted',
+    'description',
+    'doNotPublish',
+    'hide',
+    'id',
+    'kind',
+    'slug',
+    'title'
+]
+
+exercise_attributes = [
+    'allAssessmentItems',
+    'curatedRelatedVideos',
+    'description',
+    'displayName',
+    'fileName',
+    'id',
+    'kind',
+    'name',
+    'prerequisites',
+    'slug',
+    'title',
+    'usesAssessmentItems'
+]
+
+video_attributes = [
+    'description',
+    'downloadSize',
+    'duration',
+    'id',
+    'imageUrl',
+    'keywords',
+    'kind',
+    'licenseName',
+    'readableId',
+    'relatedExerciseUrl',
+    'relativeUrl',
+    'sha',
+    'slug',
+    'title',
+    'youtubeId'
+]
+
+
+def retrieve_kalite_data(lang=None, force=False) -> list:
     """
-    Retrieve the KA Lite exercises.json file in the master branch.  If
-    url is given, download from that url instead.
+    Retrieve the KA content data direct from KA.
     """
-    print("downloading exercise data")
-    if not url:
-        url = "https://raw.githubusercontent.com/learningequality/ka-lite/master/data/khan/exercises.json"
+    if lang:
+        url = "http://www.khanacademy.org/api/v2/topics/topictree?lang={lang}&projection={projection}".format(lang=lang)
+    else:
+        url = "http://www.khanacademy.org/api/v2/topics/topictree?projection={projection}"
+
+    projection = OrderedDict([
```

Comment:
Because otherwise the params get built in a different order every time - the url is different, and the cassette doesn't get checked correctly.